docs/environment-variables.adoc

```diff
@@ -51,6 +52,35 @@ Method-level configurations are visible in both `@BeforeEach` setup methods and
 Since v1.7.0, a class-level configuration means that the specified environment variables are cleared/set before and reset after each individual test in the annotated class.
 ====
 
+== `@RestoreEnvironmentVariables`
+`@RestoreEnvironmentVariables` can be used to restore changes to environment variables made directly in code.
+While `@ClearEnvironmentVariable` and `@SetEnvironmentVariable` set or clear specific variables and values, they don't allow values to be calculated or parameterized, thus there are times you may want to directly set them in your test code.
+`@RestoreEnvironmentVariables` can be placed on test methods or test classes and will completely restore all environment variables to their original state after a test or test class is complete.
+
+In this example, `@RestoreEnvironmentVariables` is used on a test method, ensuring any changes made in that method are restored:
+
+[source,java,indent=0]
+----
+include::{demo}[tag=environment_method_restore_test]
+----
+
+[NOTE]
+====
+Modifying environment variables in code requires using the Java reflection API.
+The two `@RestoreEnvironmentVariables` examples uses an https://github.com/junit-pioneer/junit-pioneer/blob/63c82ed9c417048f88c334aad5678fe1148b3d4b/src/main/java/org/junitpioneer/jupiter/EnvironmentVariableUtils.java#L35[internal JUnitPioneer utility] to do that, which can be copied to your code if needed.
```

Comment:
Suggestions welcome on this.  I wanted to include a simple example of modifying Env Vars in code, but that takes a lot of setup w/o a utility class.  But, the `EnvironmentVariableUtils` is not public, so the example is not one that a user could copy/paste.  Other ideas?  Maybe it doesn't matter if user's can directly use the example code?

Reply:
Because the approach we take is brittle, I'd prefer not to include any actual code examples that modify environment variables during test execution - we want to avoid users mistakenly thinking we support `EnvironmentVariableUtils`. 😄

Reply:
References to the util method removed.